### PR TITLE
chore: rename the prod name from ROCKs to rocks

### PR DESCRIPTION
**Pull request by cjdcordeiro**
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

Both for the code and docs, make "ROCK" lowercase (or capitalized, when applicable)

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name: rockcraft
 base: core22
-summary: A craft like experience to create ROCKS
+summary: A craft like experience to create rockS
```
Comment (linostar) [inline snap/snapcraft.yaml:3]: `rocks` not `rockS`
(but I see now that Tiago beated me to it)

File: docs/tutorials/migrate-to-chiselled-rock.rst [hunk 2]
```diff
@@ -223,22 +223,22 @@ The output should be similar to:
...
+First, import the recently created rock into Docker:
 
 .. literalinclude:: code/migrate-to-chiselled-rock/task.yaml
     :language: bash
     :start-after: [docs:skopeo-copy]
     :end-before: [docs:skopeo-copy-end]
     :dedent: 2
 
-Now inspect the Chiselled .NET Runtime ROCK the same way as it was done for the
+Now inspect the Chiselled .NET Runtime rock the same way as it was done for the
```
Comment (linostar) [inline docs/tutorials/migrate-to-chiselled-rock.rst:241]: if you agree with Tiago's note on Chiselled, then this should be changed too.

File: docs/tutorials/migrate-to-chiselled-rock.rst [hunk 3]
```diff
@@ -223,22 +223,22 @@ The output should be similar to:
     2023-04-19 15:56:00.567 :: 2023-04-19 13:56:00.148 Exported to OCI archive 'dotnet-runtime_chiselled_amd64.rock'
 
 At the end of the process, a file named ``dotnet-runtime_chiselled_amd64.rock``
-should be present in the current directory. That's your Chiselled ROCK,
+should be present in the current directory. That's your Chiselled rock,
```
[the remark this answers is not in the document]
Comment (linostar) [reply docs/tutorials/migrate-to-chiselled-rock.rst:226]: Line 217:
```
- 2023-04-19 15:56:00.567 :: 2023-04-19 13:55:59.792 Setting the ROCK's Control Data
+ 2023-04-19 15:56:00.567 :: 2023-04-19 13:55:59.792 Setting the rock's Control Data
```

**Comment by cjdcordeiro** [comment]
btw, @tigarmo the linters seem to be failing on something unrelated to this PR (?)

File: docs/explanation/rocks.rst [hunk 4]
```diff
@@ -57,13 +57,13 @@ result from the user's instructions at image build time, and that can be
 accessed by the container application at runtime.
 
 The following diagram depicts the different OCI components in the context of a
-ROCK, highlighting where the aforementioned design features (like the Pebble
+rock, highlighting where the aforementioned design features (like the Pebble
 entrypoint) fit in.
 
 .. figure:: /_static/rock_diagram.png
```
[the remark this answers is not in the document]
Comment (cjdcordeiro) [reply docs/explanation/rocks.rst:63]: nice catch. incoming commit with a fix

File: docs/tutorials/migrate-to-chiselled-rock.rst [hunk 5]
```diff
@@ -147,10 +147,10 @@ file above. You are requesting Rockcraft to install the ``libs`` slice of the
 <https://github.com/canonical/chisel-releases/blob/ubuntu-22.04/slices/>`_.
 
 
-Pack the Chiselled ROCK with Rockcraft
+Pack the Chiselled rock with Rockcraft
```
[the remark this answers is not in the document]
Comment (cjdcordeiro) [reply docs/tutorials/migrate-to-chiselled-rock.rst:150]: you're right

File: docs/tutorials/migrate-to-chiselled-rock.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (cjdcordeiro) [reply docs/tutorials/migrate-to-chiselled-rock.rst:226]: I've changed it

File: snap/snapcraft.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (cjdcordeiro) [reply snap/snapcraft.yaml:3]: :disappointed:

[the remark this answers is not in the document]
Comment (cjdcordeiro) [reply snap/snapcraft.yaml:3]: fixed

File: docs/tutorials/migrate-to-chiselled-rock.rst (hunk 2 above)
Comment (cjdcordeiro) [reply docs/tutorials/migrate-to-chiselled-rock.rst:241]: well...I'm not content writer not linguistics expert. I think cases like these are acceptable when you're referring to it as a "thing". In this case, the image is a product.
Anyway, I've applied the suggestion

**Comment by cjdcordeiro** [comment]
> btw, @tigarmo the linters seem to be failing on something unrelated to this PR (?)

right, it's being fixed in https://github.com/canonical/rockcraft/pull/439#issuecomment-1845653927

we'll rebase once that's merged

**Review by linostar — APPROVED** [review]

**Review by tigarmo — APPROVED** [review]
you guys are all plotting against me with these force-pushes, I swear :P
I'm going to assume the suggested fixes were applied

**Comment by cjdcordeiro** [comment]
> you guys are all plotting against me with these force-pushes, I swear :P I'm going to assume the suggested fixes were applied

xD you can always check what changed after your last review

**Comment by tigarmo** [comment]
but that's the feature that breaks with force pushes! The commit I reviewed isn't there anymore so Github just gives up

**Comment by cjdcordeiro** [comment]
> but that's the feature that breaks with force pushes! The commit I reviewed isn't there anymore so Github just gives up

it's [here](https://github.com/canonical/rockcraft/compare/1445d2d36fb713e081e5fde089976e4983686d31..83ed53f5eee2c431828e616e0c9e90291d455b98)

but it is true that it gets messy as it destroys history. the thing is that I don't want to squash this PR's commits. I could've rebased them after they had been approved. I only just now noticed you have the `do-not-squash` label, so I'll start using that

**Comment by tigarmo** [comment]
OK. Well, now the PR has merge commits so I don't know what a regular rebase-and-merge is going to do. Maybe you should force-push a clean set of commits? It's already approved so there's no risk of breaking review

**Comment by cjdcordeiro** [comment]
Yes I will, waiting for the tests

**Comment by cjdcordeiro** [comment]
Alright, I had to cleanup the commit history because of the previous rebase (found a typo after that).

Should be clean now. Ready to be merged once the tests pass

File: docs/how-to/code/convert-to-pebble-layer/rockcraft.yaml [hunk 6]
```diff
@@ -1,9 +1,9 @@
 name: custom-nginx-rock
 base: "ubuntu@22.04"
 version: latest
-summary: An NGINX ROCK
+summary: An NGINX rock
```
Comment (cmatsuoka) [inline docs/how-to/code/convert-to-pebble-layer/rockcraft.yaml:4]: Not the focus here, but upstream nginx call themselves "nginx" in lowercase.

**Review by syu-w — APPROVED** [review]

File: docs/how-to/code/convert-to-pebble-layer/rockcraft.yaml (hunk 6 above)
Comment (cjdcordeiro) [reply docs/how-to/code/convert-to-pebble-layer/rockcraft.yaml:4]: good point, but is it though? I've seen the 3 forms :p NGINX, Nginx and nginx xD

**Review by cmatsuoka — APPROVED** [review]
Looks good, just check if the nginx spelling is worth fixing as well.

File: docs/how-to/code/convert-to-pebble-layer/rockcraft.yaml (hunk 6 above)
Comment (cmatsuoka) [reply docs/how-to/code/convert-to-pebble-layer/rockcraft.yaml:4]: You're right, the official spelling is totally inconsistent.

**Review by lengau — APPROVED** [review]
There is [one remaining instance](https://github.com/canonical/rockcraft/blob/150460581bdab6586a7fdc59d4fcfe74c99d415a/docs/tutorials/migrate-to-chiselled-rock.rst?plain=1#L217) of "ROCK", but since that's in sample log data it's probably not something to change.

It may be worth separately updating those tutorial examples though, since Rockcraft has had quite a few UX changes in that time.

**Pull request merged**